compiler-core/src/build/package_compiler.rs

```diff
@@ -550,6 +550,24 @@ fn analyse(
                 // Register the types from this module so they can be imported into
                 // other modules.
                 let _ = module_types.insert(module.name.clone(), module.ast.type_info.clone());
+
+                // Check for empty modules and emit warning
+                let public_definitions = module
+                    .ast
+                    .definitions
+                    .iter()
+                    .filter(|def| def.is_public())
+                    .count();
```

Comment:
This one is still using the AST rather than the module interface, convert it too please